daemon/daemon.go

```diff
@@ -101,6 +101,24 @@ func NewDaemon(cfg *config.Config) *Daemon {
 		return nil
 	}
 
+	// get containerd version to make sure containerd has started
+	_, err = ctrdClient.Version(context.Background())
+	if err != nil {
+		logrus.Errorf("failed to get containerd version: %v", err)
+		return nil
+	}
+
+	if cfg.Snapshotter != "" {
+		ctrd.SetSnapshotterName(cfg.Snapshotter)
+	}
+
+	if err = checkSnapshotter(ctrd.CurrentSnapshotterName(), filepath.Join(cfg.HomeDir, "containerd/root")); err != nil {
+		logrus.Errorf("check snapshotter driver falied: %v", err)
+		return nil
```

Comment:
Do we need to return err to block daemon startup? @wangforthinker
Just double check.

Reply:
if snapshotter is checked to be invalid, pouchd should not start up in case of two kinds of snapshotter driver are both in use.

Reply:
I think we should print which driver is invalid.

I don't understand why return nil....

Reply:
Return nil means daemon startup fails.  @rudyfly

Reply:
We can only `return nil`  since `func NewDaemon(cfg *config.Config) *Daemon`. Not a good design for the function, should be `func NewDaemon(cfg *config.Config) (*Daemon, error)`. Anyway, not related to this PR

Reply:
OK, I know.